pkg/controller/releasemanager/releasemanager_controller.go

```diff
@@ -606,3 +609,36 @@ func (r *ResourceSyncer) SyncVirtualService() error {
 	}
 	return nil
 }
+
+// MarkExpiredReleases marks a Release as Expired if the Release TTL has passed,
+// and the Release is further away from any current Release than the configured buffer,
+// as sorted by GitTimestamp
+func (r *ResourceSyncer) MarkExpiredReleases() error {
+	incarnations, err := r.IncarnationList.SortedReleases()
+	log.Info("Garbage collecting releases")
+	if err != nil {
+		return err
+	}
+
+	lastCurrent := 0
+	for i, incarnation := range incarnations {
+		releaseStatus := r.Instance.ReleaseStatus(incarnation)
+		if releaseStatus.CurrentPercent > 0 {
+			lastCurrent = i
+		} else {
+			expiration := releaseStatus.GitTimestamp.Add(time.Duration(incarnation.Spec.Release.TTL) * time.Second)
+			if lastCurrent < i && i > lastCurrent+incarnation.Spec.Release.GcBuffer && time.Now().After(expiration) {
```

Comment:
lsn't `lastCurrent < i` superfluous?

Reply:
Right, this was leftover from an edge case in my first attempt. Not needed.